src/test/unit/encoders/RandomDistributedScalarEncoderTest.cpp

```diff
@@ -0,0 +1,63 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2019, David McDougall
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * --------------------------------------------------------------------- */
+
+/** @file
+ * Unit tests for the RandomDistributedScalarEncoder
+ */
+
+#include "gtest/gtest.h"
+#include <nupic/types/Sdr.hpp>
+#include <nupic/encoders/RandomDistributedScalarEncoder.hpp>
+#include <string>
+#include <vector>
+
+using namespace nupic;
+using namespace nupic::sdr;
+using namespace nupic::encoders;
+
+TEST(RDSE, testConstruct) {
```

Comment:
it would be nice to have more tests in C++, compared what you have in Py. The py tests would check here as well, but the bindings is optional, so c++ coverage should be the main focus for the next time.

Reply:
Although the bindings are option for a user running nupic, they are not optional for developers.  The py tests are part of our CI.